assets/js/base/context/cart-checkout/event-emit/reducer.js

```diff
@@ -33,19 +33,19 @@ export const actions = {
  * @param {Object} action Incoming action object
  */
 export const reducer = ( state = {}, { type, eventType, id, callback } ) => {
+	const newEvents = new Map( state[ eventType ] );
 	switch ( type ) {
 		case TYPES.ADD_EVENT_CALLBACK:
+			newEvents.set( id, callback );
 			return {
 				...state,
-				[ eventType ]: {
-					...state[ eventType ],
-					[ id ]: callback,
-				},
+				[ eventType ]: newEvents,
 			};
 		case TYPES.REMOVE_EVENT_CALLBACK:
+			newEvents.delete( id );
```

Comment:
I love how switching to `Map` removes a dependency, it seems to make the logic clearer to and reduces some code.